tests/sentry/api/test_paginator.py

```diff
@@ -599,10 +603,114 @@ def test_simple(self):
         prev_cursor = result.prev
         result = paginator.get_result(limit=3, cursor=next_cursor)
         page3_results = list(result)
-        print(page3_results)
         assert len(result) == 2
         assert page3_results[0].id == rule2.id
         assert page3_results[1].id == rule3.id
 
         result = paginator.get_result(limit=3, cursor=prev_cursor)
         assert list(result) == page1_results
+
+    def test_multi_key_order_by(self):
+        Rule.objects.all().delete()
+
+        alert_rule0 = self.create_alert_rule(name="alertrule0")
+        alert_rule1 = self.create_alert_rule(name="alertrule1")
+        assert Rule.objects.all().count() == 1
+        rule0 = Rule.objects.all().first()
+        rule1 = Rule.objects.create(label="rule1", project=self.project)
+        alert_rule2 = self.create_alert_rule(name="alertrule2")
+        alert_rule3 = self.create_alert_rule(name="alertrule3")
+        # rule2 = Rule.objects.create(label="rule2", project=self.project)
+        rule3 = Rule.objects.create(label="rule3", project=self.project)
+        # rule4 = Rule.objects.create(label="A", project=self.project)
+        rule5 = Rule.objects.create(label="a", project=self.project)
+        rule6 = Rule.objects.create(label="Z", project=self.project)
+        rule7 = Rule.objects.create(label="z", project=self.project)
+
+        alert_rule_intermediary = CombinedQuerysetIntermediary(AlertRule.objects.all(), "name")
+        rule_intermediary = CombinedQuerysetIntermediary(Rule.objects.all(), "label")
+        paginator = CombinedQuerysetPaginator(
+            intermediaries=[alert_rule_intermediary, rule_intermediary], desc=True,
+        )
+
+        result = paginator.get_result(limit=3, cursor=None)
+        assert len(result) == 3
+        page1_results = list(result)
+
+        assert page1_results[0].id == rule7.id
+        assert page1_results[1].id == rule3.id
+        assert page1_results[2].id == rule1.id
+
+        next_cursor = result.next
+        result = paginator.get_result(limit=3, cursor=next_cursor)
+        page2_results = list(result)
+        assert len(result) == 3
+        # assert page2_results[0].id == rule1.id
+        assert page2_results[0].id == alert_rule3.id
+        assert page2_results[1].id == alert_rule2.id
+        assert page2_results[2].id == alert_rule1.id
+
+        next_cursor = result.next
+        result = paginator.get_result(limit=3, cursor=next_cursor)
+        page3_results = list(result)
+        assert len(result) == 3
+        # assert page3_results[0].id == alert_rule1.id
+        assert page3_results[0].id == alert_rule0.id
+        assert page3_results[1].id == rule5.id
+        assert page3_results[2].id == rule6.id
```

Comment:
It's a little weird that this is at the end. I get why that is the case, but I think it'd make sense to sort in a case insensitive way.

```
from django.db.models.functions import Lower
MyModel.objects.order_by(Lower('myfield'))
```

Probably not necessary in this pr, maybe a follow-up.